### DISCUSSION TITLE
Issues with `regenerate_models()`

### DISCUSSION THREAD

**Discussion by elstavro** [PMEAL/OpenPNM#2804] [Q&A]
Hello,
I'm running a flow simulation from a nework that I saved to csv and I'm importing then importing it (`network_fromcsv`)

I have some issues that I believe they related to importing from csv:

- I can't print the network -- `print(pn)` gives `TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe'' `
I can print all parameters, e.g. `pn['pores.all']` and run the simulation.
I don't know why`network_from_csv()` assigns empty cells to NaN, so to be able to open it I have modified the function locally by adding `low_memory=False`

- The most important issue is that I can't run `regenerate_models()`. If I run `regenerate_models(propnames = None)` I get 
`IndexError: arrays used as indices must be of integer (or boolean) type`

![Screenshot_20230823_110553](https://github.com/PMEAL/OpenPNM/assets/111370657/4d7859a7-56e5-4f01-ac9f-e4e90f361dd4)

Any ideas? Thanks in advance!

**Comment by jgostick** [comment]
The saving and loading of networks is really only meant for data.  If you want to save the whole thing, models and all, you need to just pickle it.  If you load in some data, the thought is that you don't need the models anymore since the data is already present.  In V3 we scaled back the save/load functionality because it was trying to do too much.

**Comment by elstavro** [comment]
OK, I see, thank you!
